DiztinGUIsh/util/RomByteDataGridRow.cs

```diff
@@ -0,0 +1,598 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics.CodeAnalysis;
+using System.Drawing;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Windows.Forms;
+using Diz.Core.model;
+using Diz.Core.model.byteSources;
+using Diz.Core.model.snes;
+using Diz.Core.util;
+using DiztinGUIsh.controller;
+using JetBrains.Annotations;
+using Label = Diz.Core.model.Label;
+
+namespace DiztinGUIsh.util
+{
+    public interface IGridRow<TItem>
+    {
+        IBytesGridViewer<TItem> ParentView { get; init; }
+        Data Data { get; init; }
+        ByteEntry ByteEntry { get; init; }
+    }
+    
+    /*[AttributeUsage(AttributeTargets.Property)]
+    public class CellStyleFormatter : Attribute
+    {
+        public Func<Color?> BackgroundColorFormatter { get; }
+
+        public CellStyleFormatter(Func<Color?> bgColorFormatter)
+        {
+            BackgroundColorFormatter = bgColorFormatter;
+        }
+    }*/
+
+    [SuppressMessage("ReSharper", "UnusedMember.Global")]
+    public class RomByteDataGridRow : INotifyPropertyChanged, IGridRow<ByteEntry>
```

Comment:
this class is a little heavy, but it's awesome. whatever the properties are will be what the datagrid displays.

you can modify the grid just by swapping in a different class with a different arrangement of properties.